helpers/mysql/lib/opentelemetry/helpers/mysql.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0module OpenTelemetry
+
+module OpenTelemetry
+  module Helpers
+    # This class contains for MySQL instrumentation libraries, like mysql2 and
+    # trilogy. It is intended for use by instrumentation developers and not to
+    # be called directly by an application.
+    #
+    # To use this in your instrumentation, the `Instrumentation` class for
+    # your gem must contain configuration options for:
+    #  * `:span_name`
+    #  Example:
+    #  `option :span_name, default: :statement_type, validate: %I[statement_type db_name db_operation_and_name]`
+    #
+    # @api public
+    class MySQL
+      QUERY_NAMES = [
+        'set names',
+        'select',
+        'insert',
+        'update',
+        'delete',
+        'begin',
+        'commit',
+        'rollback',
+        'savepoint',
+        'release savepoint',
+        'explain',
+        'drop database',
+        'drop table',
+        'create database',
+        'create table'
+      ].freeze
+
+      QUERY_NAME_REGEX = Regexp.new("^(#{QUERY_NAMES.join('|')})", Regexp::IGNORECASE)
+
+      # This is a span naming utility intended for use in MySQL database
+      # adapter instrumentation.
+      #
+      # @param sql [String] The SQL statement for the span.
+      # @param operation [String] The database operation.
+      # @param database_name [String] The name of the database.
+      # @param config [Hash] The user's configuration for the database adapter.
+      #  Desired keys:
+      #    * `:span_name` => A symbol describing the type of name desired. Expected options are `:statement_type`, `:db_name`, and `:db_operation_and_name`. A nil or unknown `:span_name` will return 'mysql' as the span name
+      # @return [String] The span name.
+      # @api public
+      def self.database_span_name(sql, operation, database_name, config)
+        case config[:span_name]
+        when :statement_type
+          extract_statement_type(sql)
+        when :db_name
+          database_name
+        when :db_operation_and_name
+          db_operation_and_name(operation, database_name)
+        end || 'mysql'
+      end
+
+      # @api private
+      def self.extract_statement_type(sql)
+        QUERY_NAME_REGEX.match(sql) { |match| match[1].downcase } unless sql.nil?
+      rescue StandardError => e
+        OpenTelemetry.logger.debug("Error extracting sql statement type: #{e.message}")
```

Comment:
Was this a debug error in other places? 

Should this use `handle_error` instead?

Reply:
In MySQL2, it was a debug-level log. For Trilogy, it was an error-level log. I'm happy to use`handle_error` instead.

Reply:
@arielvalentin  - This has been replaced with a `handle_error` call: f46b1f7d9b8a1e073c200afa5f6676f379b2a34a 

Let me know if there are any further changes you'd like me to make. I'll rebase the commit after I get your approval.

Reply:
> In MySQL2, it was a debug-level log. For Trilogy, it was an error-level log.

Good old copy/paste modify poorly. _Spoiler alert: IT WAS ME_